predict-idlab/plotly-resampler
# :package: serialization support +  :level_slider: update OS & python version in test-matrix

Should fix #85 

In shot this PR adds the following contributions;
* Add serialization to plotly-resampler figures
  => serialize the plotly-resampler figure with its `hf_data` and other plotly-resampler properties, allowig pickling & (deep)copy!
  (the support for pickling allows to start processes that show a plotly-resampler figure via spawning)
* Extend the test-matrix with (1) Windows & macOS, (2) python 3.10

This PR does;
* #92 
  - [x] support figure as a dict as input for our wrappers (e.g. `FigureResampler(fig.to_dict())` with `fig` a `go.Figure`)
    => this is in line with the vanilla plotly behavior (see `BaseFigure` constructor documentation)
  - [x] propagate `_grid_str` in our composable constructors
  - [x] test the above
* #93
  - [x] disable `test_fwr_time_based_data_s`
* add serialization support
  => this allows starting a `multiprocessing.Process` on Windows & Mac-OS (as these use spawning) https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
  (note that this already worked on Unix as for such OS forking is used)
  - [x] support serialization of plotly-resampler figures
  - [x] test serialization when pickling
  - [x] test serialization when (deep)copying 
* #97 
  - [x] :bug: fix that inline show dash has correct height when not set in layout
  - [x] :sparkles: call `show_dash` for `FigureResampler` on `_ipython_display_`
  - [x] add `show_dash_kwargs` to `FigureResampler` constructor 
  - [x] test the above
  - [ ] update docs for `show_dash_kwargs` 
* update tests for (1) Windows & Mac-OS, and (2) Python 3.10
  - [x] fix selenium tests for Windows & Mac-OS (#95)
  - [x] add Python 3.10 (#96)
